src/pages/PhotoPage.tsx

```diff
@@ -9,32 +9,62 @@ type PhotoPageProps = {
 }
 
 function PhotoPage({ toggleHeader }: PhotoPageProps) {
+  const [windWidth, setWindWidth] = useState(window.innerWidth)
+  function isNavShowing() {
+    setWindWidth(window.innerWidth)
+  }
+
   useEffect(() => {
     toggleHeader(false)
-    return () => toggleHeader(true)
-  }, [toggleHeader])
+    window.addEventListener("resize", isNavShowing)
+    if (windWidth > 900) {
+      toggleHeader(true)
+    } else {
+      toggleHeader(false)
+    }
+    return () => {
+      toggleHeader(true)
+      window.removeEventListener("resize", isNavShowing)
+    }
+  }, [toggleHeader, windWidth])
 
-  const { photoId } = useParams<{ photoId: string }>()
-  const { photoQuery } = useApi()
+  const { albumId, photoId } = useParams<{ albumId: string; photoId: string }>()
 
-  const { data, error, isError, isLoading } = photoQuery(photoId)
-  console.log(data)
+  const { photoInAlbum, photoQuery } = useApi()
+  if (!albumId) {
```

Comment:
You can reduce the code inside the if to just the API query, the rest will be the same:

```typescript
let data, error, isError, isLoading
if (albumId) {
    { data, error, isError, isLoading } = photoInAlbumQuery(albumId, photoId)
} else {
    { data, error, isError, isLoading } = photoQuery(photoId)
}

if (isLoading) {
  return <h1>Info is loading</h1>
}
// etc
```

Reply:
Absolutely, this way is much more cleaner!